main.py

```diff
@@ -0,0 +1,224 @@
+from pydantic import BaseModel, Field
+from datetime import datetime
+from fastapi.responses import HTMLResponse
+from fastapi.templating import Jinja2Templates
+
+
+DATABASE_URL = "sqlite:///home_work/home_work_6/my_database.db"
+database = databases.Database(DATABASE_URL)
+
+metadata = sqlalchemy.MetaData()
+
+# Таблица должна содержать следующие поля: id (PRIMARY KEY), имя, фамилия, адрес электронной почты и пароль.
+users = sqlalchemy.Table("users", metadata,
+                         sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+                         sqlalchemy.Column("name", sqlalchemy.String(32)),
+                         sqlalchemy.Column("lastname", sqlalchemy.String(32)),
+                         sqlalchemy.Column("email", sqlalchemy.String(128)),
+                         sqlalchemy.Column("passwd", sqlalchemy.String(32)),
+                         )
+# Таблица товаров должна содержать следующие поля: id (PRIMARY KEY), название, описание и цена.
+products = sqlalchemy.Table("products", metadata,
+                            sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+                            sqlalchemy.Column("product_name", sqlalchemy.String(32)),
+                            sqlalchemy.Column("description", sqlalchemy.String(128)),
+                            sqlalchemy.Column("price", sqlalchemy.Float),
+                            )
+# Таблица заказов должна содержать следующие поля: id (PRIMARY KEY), id пользователя (FOREIGN KEY),
+# id товара (FOREIGN KEY), дата заказа и статус заказа.
+orders = sqlalchemy.Table("orders", metadata,
+                          sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+                          sqlalchemy.Column("id_user", sqlalchemy.ForeignKey('users.id')),
+                          sqlalchemy.Column("id_product", sqlalchemy.ForeignKey('products.id')),
+                          sqlalchemy.Column("order_date", sqlalchemy.DateTime),
+                          sqlalchemy.Column("status", sqlalchemy.Boolean),
+                          )
+
+engine = sqlalchemy.create_engine(DATABASE_URL, connect_args={'check_same_thread': False})
+metadata.create_all(engine)
+
+app = FastAPI()
+templates = Jinja2Templates(directory="./home_work/home_work_6/templates")
+
+
+@app.on_event("startup")
+async def startup():
+    await database.connect()
+
+
+# два метода,
+# для connect и disconnect БД ! (при запуске создается БД без таблиц, если их нет)
+
+
+@app.on_event("shutdown")
+async def shutdown():
+    await database.disconnect()
+
+
+class UserIn(BaseModel):
+    name: str = Field(max_length=32)
+    lastname: str = Field(max_length=32)
+    email: str = Field(max_length=128)
+    passwd: str = Field(max_length=32)
+
+
+class User(BaseModel):
+    id: int
+    name: str = Field(max_length=32)
+    lastname: str = Field(max_length=32)
+    email: str = Field(max_length=128)
+    passwd: str = Field(max_length=32)
```

Comment:
Модели почти идентичны, а значит, одна может наследоваться вот другой.
```suggestion
class User(UserIn):
    id: int
```